### PR TITLE
hardwires the earliest fulfilment date for weekly to the 24th

### PR DESCRIPTION
**Pull request by AWare**
hardwires the earliest fulfilment date for weekly to the 24th 
(and renames the method to calculate it in the checkoutservice)

### CODE REVIEW

**Comment by johnduffell** [comment]
but why?

**Comment by AWare** [comment]
https://trello.com/c/lA03nR4C/197-guardian-weekly-hard-code-selected-date-to-friday-24th-till-fulfilment-report-has-gone-out

**Comment by johnduffell** [comment]
actually it doesn't hard code it, it puts a limit on how early it can be

**Comment by johnduffell** [comment]
sounds good, although I feel like we should just change the fulfilment delay to be 8 days instead of 7 😆 👍

**Comment by AWare** [comment]
@jacobwinch

**Comment by jacobwinch** [comment]
However, we need some kind of reminder to remove the references to the 24th and to un-comment the tests.

**Comment by jacobwinch** [comment]
(Since we don't want to revert this pull request, as then we'll lose your refactoring)

File: app/services/CheckoutService.scala [hunk 1]
```diff
@@ -44,16 +44,14 @@ object CheckoutService {
   def paymentDelay(in: Either[PaperData, DigipackData], zuora: ZuoraProperties)(implicit now: LocalDate): Days = in.fold(
     p => Days.daysBetween(now, p.startDate), d => zuora.gracePeriodInDays.plus(zuora.defaultDigitalPackFreeTrialPeriod)
   )
+  def nextFriday(d: LocalDate):LocalDate = d match {
```
Comment (johnduffell) [inline app/services/CheckoutService.scala]: much clearer logic!!! :1st_place_medal:

**Pull request merged**